File: proxy/tests/script_engine/trailer_length.js
/// <reference path="../../script_libs/js/index.d.ts" />
/** @type {Extension} */
const trailer_has = {
  request(flow) {
    const t = flow.request.trailers;
    if (t.length == 12) {
      t.clear();
    }
  },
  response(flow) {
    const t = flow.response.trailers;
    if (t.length == 12) {
      t.clear();
    }
  }
}
globalThis.extensions = [trailer_has];

File: proxy/tests/script_engine/trailer_length.py
from roxy import Extension


class LengthTrailers(Extension):
    def request(self, flow):
        t = flow.request.trailers
        if len(t) == 12:
            t.clear()

    def response(self, flow):
        t = flow.response.trailers
        if len(t) == 12:
            t.clear()


Extensions = [LengthTrailers()]
